core/src/main/java/io/questdb/griffin/engine/functions/columns/SymbolColumn.java

```diff
@@ -58,7 +58,8 @@ public CharSequence getSymbolB(Record rec) {
     @Override
     public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) {
         this.symbolTable = symbolTableSource.getSymbolTable(columnIndex);
-        assert !symbolTableStatic || symbolTable != null;
+        //non-static symbol table must be non-null
+        assert symbolTableStatic || symbolTable != null;
```

Comment:
I think it's the other way around, when static symbol table is static it means non-null should exist. `getSymbolTable` is synonym to `getStaticSymbolTable`

src/main/java/io/questdb/griffin/engine/functions/eq/EqSymStrFunctionFactory.java